XMonad/Hooks/WallpaperSetter.hs

```diff
@@ -198,18 +200,18 @@ applyWallpaper parts = do
   layers <- liftIO $ mapM layerCommand parts
   let basepart ="convert -size "++show vx++"x"++show vy++" xc:black "
       endpart =" jpg:- | feh --no-xinerama --bg-tile --no-fehbg -"
```

Comment:
There's still this in here; will this interfere with using non-jpg image formats?

Reply:
I am testing for this now.

Reply:
Sadly, I found out that this can not be altered. It can only be set to one among `png` `jpg` .... 
The operation is basically this
1. if there are multiple monitors, convert the two pictures into one big picture, then use `feh` to display it. For example `1920x1080` + `2560x1440` -> `4480x1440`

Reply:
convert -size 4480x1440 xc:black \( '<path to file>'  -scale 2560x1440^ -gravity center -extent 2560x1440 +gravity \) -geometry +1920+0 -composite   \( '<path to file>'  -scale 1920x1080^ -gravity center -extent 1920x1080 +gravity \) -geometry +0+0 -composite  jpg:- | feh --no-xinerama --bg-tile --no-fehbg -

Reply:
So I think besides adding `defWPName`, there isn't much that can be done here.

Reply:
Will this work even with the `jpg:-` part?  Or do you need to set it to `png:-` when `defWPName'` (n.b.: perhaps `defWPNamePng`) is used?

Reply:
Actually it is ok to use either `png:-` or `jpg:-`. For example, when specifying `wallpaperbasedir` and let the random function pick one image, it is actually using `png` and convert it to `jpg` (in my case).

Reply:
> Will this work even with the `jpg:-` part? Or do you need to set it to `png:-` when `defWPName'` (n.b.: perhaps `defWPNamePng`) is used?

I think this is really an good suggestion. Maybe I can rename both defWPName and defWPName' to

Reply:
> Will this work even with the `jpg:-` part? Or do you need to set it to `png:-` when `defWPName'` (n.b.: perhaps `defWPNamePng`) is used?

`jpg:-` and `png:-` only specify the final format. Actually as the example I have given, you can mix jpg and png and finally say that the result is a png file.

Reply:
> I think this is really a good suggestion. Maybe I can rename both `defWPName` and `defWPName'` to `defWPNameJpg` and `defWPNamePng`

I think `defWPName` should stay, for backwards compatibility.  `defWPName' -> defWPNamePng` sounds good though!

Reply:
Thanks, I have added it.